README.md

```diff
@@ -1,129 +1,108 @@
+output "nitro_dev_instance_ip" {
+  description = "The public IP address of the instance"
+  value       = module.nitro_dev.instance_public_ip
+}
+```
 
-From the base environment, create and activate (update) a Python environment
-specified in `environment.yaml`
+For the "key_pair_name", you will need to have a key pair already created in
+AWS and the private key on your local machine.  Below, we will refer to the
+location of this key on your local machine as `<path-to-your-pem-file>`.
 
-    mamba env update -n bky-ned -f environment.yaml
-    conda activate bky-ned
+You will likely want to create an output with the ip address of the
+instance that is created so that you can use it later in ansible playbooks.
 
-Install all the python dependencies
+Let's create the infrastructure.
 
-    poetry install
+```bash
+cd terraform
+terraform init
+terraform plan -out tfplan
+terraform apply tfplan
+```
 
-Make sure that everything is set up properly for development by running:
+Upon successful completion of the `apply` you will see the ip address of the
+instance that was created. You can copy it for later or put it into an
+environment variable like so:
 
-    make test
+```bash
+ip=$(terraform output nitro_dev_instance_ip)
+```
 
-When you want to clean up run:
+Also, if you hop over to the aws console, you will see that your instance is
+deployed. You can also get the ip address from the console. Either way, let's
+clean up the plan we used and move on to the next step.
 
-    conda deactivate
-    conda remove -n sequencer --all
-    make veryclean
+```bash
+rm tfplan
+cd ..
+```
 
-While you can run unit tests without configuring the NED, you will need to
-[Configure NED](#configuring-ned) to run `make test-live` (live tests) or use the
-application.
+Let's use our `init-nitro` playbook to minimally set up a system.
 
-<a name="configuring-ned"></a>
-###  Configuring NED
+```bash
+ansible-playbook \
+    -i ansible/inventories/dev.yml \
+    -e ip=$ip \
+    --key-file <path-to-your-pem-file> \
+    ./ansible/playbooks/init-nitro.yml
+```
 
+Personally, I like to use the nix package manager in my development
+environments.  If you would like to use nix, you can run the following playbook.
 
-To set up, first, we will need a place to store NED secrets.  Let's put that in our
-home directory.
+```bash
+ansible-playbook \
+    -i ansible/inventories/dev.yml \
+    -e ip=$ip \
+    --key-file <path-to-your-pem-file> \
+    ./ansible/playbooks/init-nix.yml
+```
 
-    mkdir -p $HOME/secrets/ned
+Now, you can ssh into the instance and start hacking away on your project
+with nitro.
 
-Next, you will need AWS credentials in a CSV. (Creating credentials for Amazon
-is well documented online or ask internally if you need a hand.) Here, we will
-assume the file is called `aws--bob-dev.csv`.  Put the creds file in your
-secrets folder (but not necessarily the NED secrets folder).  For example:
+```bash
+ssh -i <path-to-your-pem-file> ec2-user@$ip
```

Comment:
This did not work for me, because the way we're setting up the `ip` variable makes it contain the quotes around the ip. Here's the error I got:
```bash
ssh: Could not resolve hostname "18.204.220.73": Name or service not known
```

Reply:
Also, we may want to add something to ansible to use a newer image, or to run updates on creating. After I log in by passing in the IP, I get:
```bash
$ ssh -i "~/.ssh/mwittie-dev.pem" ec2-user@18.204.220.73   
Last login: Wed Jul 17 17:05:24 2024 from 66.9.181.104

       __|  __|_  )
       _|  (     /   Amazon Linux 2 AMI
      ___|\___|___|

https://aws.amazon.com/amazon-linux-2/
46 package(s) needed for security, out of 75 available
Run "sudo yum update" to apply all updates.
```